### DISCUSSION TITLE
Retiring the Hub on June 6th, 2024

### DISCUSSION THREAD

**Discussion by lossyrob** [microsoft/PlanetaryComputer#347] [Announcements]
> [!NOTE]
> The Planetary Computer Hub will be retired on the 6th of June 2024.
 
This is a notice that __the Planetary Computer Hub will be retired on the 6th of June 2024__. Please note that this change only affects the Planetary Computer Hub; the Planetary Computer Data and APIs will remain available and unchanged.

Microsoft has recently tightened security requirements across all systems ([Prioritizing security above all else - The Official Microsoft Blog](https://blogs.microsoft.com/blog/2024/05/03/prioritizing-security-above-all-else/)). Implementing these new security protocols requires changes to the Planetary Computer architecture, which the Hub is not compatible with. We want to reassure you that there have been no known security incidents within the Planetary Computer Hub. We are required to take these steps to comply with enhanced company-wide security standards. As a Preview product (see our [Terms of Use](https://planetarycomputer.microsoft.com/terms)), we must adhere to these stringent requirements and deadlines despite the resulting disruption. We do apologize for any inconvenience this retirement will cause.

We will send a notice to all registered users to download any needed data from their Planetary Computer Hub accounts. We will also be updating this discussion with instructions on how to retrieve your Hub home directory after retirement, in case you miss the deadline.

The retirement process will proceed as follows:
-	During the weekend of May 25th – 26th, the Hub will transition to a new DNS name at https://planetarycomputer-hub.microsoft.com/ . Users can log into the Hub using their same Microsoft credentials at the new domain.
-	On June 6th, the Hub will be retired and will no longer be accessible.
-	Users with data still in the system after that point will need to contact [planetarycomputer@microsoft.com](mailto:planetarycomputer@microsoft.com) to retrieve data. We will service these requests as quickly as we can but cannot guarantee prompt access to data after this time. Please take the time before June 6th to download the data you need from your account.

__Please download all data from your Planetary Computer Hub home directory before Friday, June 6th, 2024.__

The easiest way to download your files is to log into the Hub and use the Jupyterlab UI. For a small number of files, you can just use the file picker on the left-hand side.

For many files (including your entire home directory) you can:

1. Start a notebook server
2. Start a new terminal in Jupyterlab
3. Make a ZIP archive of your home directory (or whatever important directories you have in there):
   
```
# Use '.' to zip the entire Home directory.
# Replace '.' with specific directories if you want.
zip -r hub.zip .
```

4. Download that `hub.zip` using the Jupyterlab UI

Reach out to [planetarycomputer@microsoft.com](mailto:planetarycomputer@microsoft.com) with any questions or concerns.

If you need alternative platforms to the Hub, we suggest the following:
-	[Microsoft Fabric](https://learn.microsoft.com/en-us/fabric/data-engineering/how-to-use-notebook)
-	[Azure Machine Learning](https://learn.microsoft.com/en-us/azure/machine-learning/how-to-run-jupyter-notebooks?view=azureml-api-2)
-	[GitHub Codespaces](https://docs.github.com/en/codespaces/developing-in-a-codespace/getting-started-with-github-codespaces-for-machine-learning)
-	[Deploy your own Hub](https://planetarycomputer.microsoft.com/docs/concepts/hub-deployment/)

Microsoft launched the Hub in April of 2021, and since then has provided tens of millions of dollars of free compute to our user base. We thank all of you for leveraging the Planetary Computer Hub for your work and encourage you to continue to leverage the over 50 PB of data and APIs we continue to host in the Planetary Computer. We are looking forward to offering more ways to collaborate on your work together in the future.

**Comment by BnJam** [comment]
> the Planetary Computer Data and APIs will remain available and unchanged.

Can you confirm that these API's do align with Microsofts security initiatives? 

As beneficial and useful as the compute has been for so many scientists across the world, the availability of this data repository has been more beneficial, in my opinion. Having STAC accessible EO data is enabling to so many individuals, groups, and initiatives that [Microsoft  aligns with](https://www.microsoft.com/en-us/corporate-responsibility/sustainability) that it would be extremely disappointing of Microsoft to also shutter on it. 

Understandably, Microsoft is a for-profit company and there is always the risk that such data and API's can/will be removed or retired; so, I must ask the above question.

**Reply by lossyrob** [reply]
Great question and thank you for the feedback. There is no anticipated impact on the availability of our data and APIs at this time, and we have no intentions in changing the availability of our data or other services due to these requirements.

**Reply by krishnaglodha** [reply]
So it means we'll still be able to stream data using stac end points and planetary-computer python package right?

**Reply by TomAugspurger** [reply]
> So it means we'll still be able to stream data using stac end points and planetary-computer python package right?

Correct (the `planetary-computer` package just gets you URLs with SAS tokens. The actual reads happen from Azure Blob Storage).

**Comment by pkopparla** [comment]
I just want to add that I find Planetary Computer very useful for my work, and I hope that it will not be retired in the future.

**Comment by songhan89** [comment]
It’s good to know that the Data and APIs will remain. Anyway, just want to take the opportunity to thank the folks who setup this up. It has been immensely helpful for the geoscience and climate science community.

**Comment by lukesdm** [comment]
Oh, this is such a shame. The Hub really lowered the barrier for working with this data. And it was a real success story for Open Source geo software.

More personally, we have ongoing projects using this, and two weeks to procure resources and figure out how to set up our own Hub and Dask infra is painful to think about.

**Comment by robintw** [comment]
Firstly, thank you for providing the Hub for so long. It's been amazing.

I'm really disappointed by the retirement, but I had always imagined this day would come (though I assumed it would be because of escalating costs rather than security). It concerns me now that we don't have a free place to demo the processing side of the STAC-based, open-source geo workflow - I imagine a lot of people will move to Google Earth Engine, which is a shame because of the lack of flexibility and non open-source nature of the tools provided there.

Are there any plans to provide any other method of processing geo data through Planetary Computer? The STAC APIs etc are great, but without processing ability it feels rather 'empty'.

**Reply by m-mohr** [reply]
I'm wondering whether a more restrictive API for processing such as the openEO API on top of MS PC would work with their security requirements. That running arbitrary user code on a platform may incur some risk is clear, but openEO only allows to run more abstract data cube based processing and as such can be controlled. The dask/xarray or Spark implementation(s) must probably go through a security review, but it would give back some processing capabilities close to the data...

**Reply by TomAugspurger** [reply]
> It concerns me now that we don't have a free place to demo the processing side of the STAC-based, open-source geo workflow

We are still fortunate to have [mybinder.org](https://mybinder.org/) (though the funding there is a perennial challenge). I'm pretty excited by pyodide and [jupyterlite](https://jupyterlite.readthedocs.io/en/stable/quickstart/index.html), which pairs nicely with the tiler / `data` API. Though neither of these is a replacement for the large scale workflow. Those will still be best handled by one of the alternative compute options mentioned elsewhere.

**Reply by robintw** [reply]
Good point, I guess relatively limited workflows (for demo etc) can be run on mybinder.org. I'll definitely miss the free hosted Dask offering - but there are other potential (paid) options for that.

**Reply by alexgleith** [reply]
We ran a [workshop](https://github.com/auspatious/cloud-native-geospatial-eo-workshop) using GitHub Codespaces the other week and it worked really well. Not sure how it goes with scaling up to lots of memory, though.

**Comment by mrocklin** [comment]
Hopefully it's ok to mention here that [Coiled](https://coiled.io) exists, and can manage large scale Dask/Xarray workflows in Azure accounts today.  (we added Azure support a few months ago)

Coiled ...

- is a cloud service maintained by a bunch of Dask maintainers
- has a generous free tier (10,000 CPU hours per month managed in your account) 
- has loads of current geospatial customers on it today that are *very* happy after switching from managing their own infrastructure
- has fun features like environment synchronization, historical tracking of jobs, metrics dashboards, user/cost management, etc..

If people are looking for something to transition to we're here, and pretty good at managing large scale Dask/Xarray workloads.

- https://coiled.io/geo
- http://coiled.io/start
- [Two minute intro video](https://youtu.be/SfPzmIqMKp0)

Sorry for the advertisement.  Hopefully it's relevant to folks here though.

**Reply by TomAugspurger** [reply]
You beat me to it :)

**Reply by mrocklin** [reply]
Any interest in working on a migration guide together?  (No pressure.  I'm hoping that this is pretty easy)

**Reply by TomAugspurger** [reply]
Looks like https://docs.coiled.io/user_guide/usage/notebooks/index.html would be the closest analog.

In theory, the migration would be something like

- Install a local Python environment like https://github.com/microsoft/planetary-computer-containers/tree/main/python (or whatever you need. Package customization was *not* a strong point of the Hub)
- `coiled notebook start --sync` in whatever directory you're working in (all of our examples are at https://github.com/microsoft/PlanetaryComputerExample)

Some of those examples (the Dask ones in particular) would probably be nicer with a local Python kernel and the Dask cluster in Azure, but most are written with the assumption that the compute is in the same region as the data.

**Reply by mrocklin** [reply]
> Some of those examples (the Dask ones in particular) would probably be nicer with a local Python kernel and the Dask cluster in Azure, but most are written with the assumption that the compute is in the same region as the data.

Yeah, while people are attracted to notebooks it's rarely the right final solution for users.  Almost everyone (except maybe interactive GPU users) ends up using Dask clusters from local machines.  Maybe the thing to do here is to experiment a bit together, find a good UX, and then write up that experience?

**Comment by jgomezdans** [comment]
While this is disappointing, it's always the risk with these offerings. One thing that would be very appreciated would be a clear simple guide to set things up in some of the alternative platforms to the Hub that you suggest, with fairly opaque pricing and limitations.

**Comment by krishnaglodha** [comment]
I’d love to help 😊

Thanks & Regards,
Krishna G. Lodha
Founder
office email - ***@***.***
personal email - ***@***.***

**Comment by gilbertocamara** [comment]
Dear Robin et al.,

First and foremost, praise is due to MPC team and to MS for building it. They deserve significant recognition for their dedication and the quality of their work. I extend my personal appreciation to Tom for his outstanding contributions! 

While recognising that most Planetary Computer users prefer Python, it is worthwhile mentioning that the R environment is arguably better served with robust packages that can be used in connection with MPC. They also work with other cloud services such as AWS, Copernicus and Digital Earth Africa. Examples include:

(a) Combining “rstac” with “gdalcubes” allows creating data cubes from data available on cloud services;
(b) The “sits” package offers an end-to-end TRL-9 solution for land use and land cover classification. 

Another noteworthy mention is openEO. Currently, openEO is limited in its functionality for machine learning and deep learning. However, there's good news on the horizon. The EC-funded Open Earth Monitor project is actively developing a sits backend for openEO, which will significantly enhance its capabilities.

On a broader note, the potential discontinuation of the Planetary Computer Hub is both understandable and concerning. As a long-term player, I've consistently argued that Earth observation is not profitable. It's a global public good. We increasingly require Earth observation as a public service to address the climate crisis. 

I am concerned that someone high up in the Azure team might have decided to replace the current free Hub with an enhanced proprietary version as a viable business model. Such a plan is unlikely to succeed. In my 45 years of experience, I have seen repeated failures to make money by selling EO data and/or services. The Landsat privatisation in the 1990s, the French SPOT programme, Envisat-1, TerraSar, COSMOS/Skymed, IKONOS. Private companies came and went: ERDAS, PCI, Infoterra. The list goes on. In all cases, these programmes had to be rescued by the public purse, or they died. Even Planet is more than USD 150 million in red. 

External IT managers find it difficult to understand the challenge of making money from EO. Perhaps some Azure managers think that EO is like GIS. Quite the opposite. The basis of GIS has not changed in 20 years. Vector operations are eternal. The object-relational database is stable. That is the reason why ESRI thrives, and we have standards such as Simple Features SQL. 

EO is an evolving technology in a permanent state of flux. A few current challenges: hyperspectral data, multi-polarized SAR, LIDAR, OBIA, and many open questions in machine learning. It is impossible for a commercial vendor to keep track of it all. Will MS be able to design a commercial Hub that attracts interest and makes money? Welcome to the EO System Designers Club!  Past experience teaches us that designing a good API for EO is tough. Few successes, and many failures. Many EO APIs are clunky and cumbersome. MS has the programming skills, but will they have the required experience and vision to succeed where so many others have failed? 

Whatever the case, the EO community needs more joint efforts to develop good tools. These tools should be usable across platforms and thus resilient to the possible demise of privately run services. The community must advocate for creating and maintaining public EO cloud services. In the long run, public services are the most consistent way to use EO to address the immense challenges humanity is facing. 

Best regards
Gilberto
============================
Prof Dr Gilberto Camara
Senior Researcher
National Institute for Space Research (INPE), Brazil
https://gilbertocamara.org/
=============================

**Reply by kbgg** [reply]
I'd say it's pretty likely that this is a response to [Satya's memo](https://www.theverge.com/24148033/satya-nadella-microsoft-security-memo) earlier this month and nothing more.

**Comment by moradology** [comment]
Sad to see this. OS GIS owes you (and the team) a deep debt of gratitude for everything that's been enabled by the work. 🫡

**Comment by TomAugspurger** [comment]
We'll have a bit of downtime this evening as I do some maintenance on the Hub, moving the DNS record. Follow along at https://github.com/microsoft/PlanetaryComputer/issues/352.

**Reply by TomAugspurger** [reply]
We're back up at our new home https://planetarycomputer-hub.microsoft.com/ (until the final retirement on the 6th).

Ping me if you have any issues.

**Reply by PanaderoCF** [reply]
Hi Tom! I am trying to create a cluster using dask gateway, but after the update this is not working anymore. I try the following code that used to work before the update:

`gateway = dask_gateway.Gateway()
cluster_options = gateway.cluster_options()
cluster = gateway.new_cluster(cluster_options)
cluster.adapt(minimum=1, maximum=100)
cluster`

I receive the following errors: 

`gaierror: [Errno -2] Name or service not known`

`ClientConnectorError: Cannot connect to host proxy-http:8000 ssl:default [Name or service not known]`

Do you know what can be the problem and maybe help me out with it? I am desperate to perform some last calculations on the Planetary Computer.

Best regards,
Floor

**Reply by TomAugspurger** [reply]
Thanks for the report. https://github.com/microsoft/planetary-computer-hub/pull/112 should fix this. I'll get it deployed shortly.

**Reply by TomAugspurger** [reply]
Should be all fixed now.

**Comment by daviddeltares** [comment]
Hi,
It seems like the connection between my Visual Studio Code and the Planetary Computer Hub has been broken because of the new domain  https://planetarycomputer-hub.microsoft.com/. 
However, I am having trouble setting the connection up again as the documentation page regarding connecting VS Code and the planetary computer does not seem to be working anymore (https://planetarycomputer.microsoft.com/docs/overview/ui-vscode/).
Could you please post the instructions or get the page back up, thank you in advance!

Kind regards,
David

**Reply by TomAugspurger** [reply]
You can see the old version at https://github.com/microsoft/PlanetaryComputerDataCatalog/blob/2023.2.0/docs/overview/ui-vscode.md.

FWIW, I was having trouble with that method even before the URL moved, but that might have been user error on my part. You'll want to replace `https://pccompute.westeurope.cloudapp.azure.com/compute` with `https://planetarycomputer-hub.microsoft.com/`.

**Reply by lukesdm** [reply]
Hi David, did you get this working again?

**Reply by daviddeltares** [reply]
Unfortunately not, when following the instructions and putting in the following URL:
 https://planetarycomputer-hub.microsoft.com/user/my_username/?token=xxxx
it gives "Invalid URL Specified".
I have tried some other combinations but I did not get it to work, would love to hear if someone did manage to.

**Reply by lukesdm** [reply]
Ok, thanks. I get the same response. But I never tried it before the switch, so wasn't sure if it was just a mistake at my end.
[This issue](https://github.com/microsoft/vscode-jupyter-hub/issues/36) hints at some problem with the site cert., and output from my console also suggests this:

```
13:32:55.940 [error] Error sending request to https://planetarycomputer-hub.microsoft.com/user/<username>/hub/api FetchError: request to https://planetarycomputer-hub.microsoft.com/user/<username>/hub/api failed, reason: unable to verify the first certificate
    > at ClientRequest.<anonymous> (c:\Users\<user>\.vscode\extensions\ms-toolsai.jupyter-hub-2023.10.1003012305\dist\extension.node.js:63581:18)
    > type = system
    > errno = UNABLE_TO_VERIFY_LEAF_SIGNATURE
    > code = UNABLE_TO_VERIFY_LEAF_SIGNATURE
13:32:56.151 [error] Error sending request to https://planetarycomputer-hub.microsoft.com/hub/api FetchError: request to https://planetarycomputer-hub.microsoft.com/hub/api failed, reason: unable to verify the first certificate
    > at ClientRequest.<anonymous> (c:\Users\<user>\.vscode\extensions\ms-toolsai.jupyter-hub-2023.10.1003012305\dist\extension.node.js:63581:18)
    > type = system
    > errno = UNABLE_TO_VERIFY_LEAF_SIGNATURE
    > code = UNABLE_TO_VERIFY_LEAF_SIGNATURE
13:32:56.336 [error] Error sending request to https://planetarycomputer-hub.microsoft.com/hub/api FetchError: request to https://planetarycomputer-hub.microsoft.com/hub/api failed, reason: unable to verify the first certificate
    > at ClientRequest.<anonymous> (c:\Users\<user>\.vscode\extensions\ms-toolsai.jupyter-hub-2023.10.1003012305\dist\extension.node.js:63581:18)
    > type = system
    > errno = UNABLE_TO_VERIFY_LEAF_SIGNATURE
    > code = UNABLE_TO_VERIFY_LEAF_SIGNATURE
13:32:56.337 [error] Failed to determine base url for https://planetarycomputer-hub.microsoft.com/user/<username>/?token=<token> Error: Unable to determine base url of Jupyter Hub Server
    > at getJupyterHubBaseUrl (c:\Users\<user>\.vscode\extensions\ms-toolsai.jupyter-hub-2023.10.1003012305\dist\extension.node.js:64314:9)
```

I guess this is too late to hope for a fix though?

**Reply by TomAugspurger** [reply]
Mmm, all the stuff around certificates is one thing that changed with this switch. SSL termination was (I think) moved out of the Hub and into Application Gateway.

Strangely(?) my browser is just fine with the certs being used. But this `openssl`  command is able to surface the same error message

```
openssl s_client -showcerts  planetarycomputer-hub.microsoft.com:443 </dev/null
...
Verification error: unable to verify the first certificate
```

I don't know enough about how SSL works to say for sure what's going on here :/

**Reply by lukesdm** [reply]
Thanks Tom. With that info I was able to get further. It seems the JupyterHub extension is using a different SSH client which doesn't like the cert (Powershell in Windows also works for me). According to [this answer](https://github.com/microsoft/vscode-jupyter-hub/issues/36#issuecomment-2048412441) in other thread, I can bypass the cert check in VSCode with an env variable. This lets me connect to MSPC and create or connect to existing kernels. But when I try to execute a cell, I get prompted for another password. I guess this would be an OAuth token (?). Not sure how to get past that.

**Reply by TomAugspurger** [reply]
Mmm that one I don't know. I would have expected all the auth to be done by the time you get to that point.

**Reply by lukesdm** [reply]
No worries, thanks anyway.

**Comment by cmosig** [comment]
What's the correct URL to apply for an API key now? https://planetarycomputer.microsoft.com/account/request is dead.

**Reply by TomAugspurger** [reply]
@cmosig what endpoint are you trying to get that key for? There shouldn't be any that require one now.

**Reply by cmosig** [reply]
Is there no increased rate-limit for users with an API key anymore when accessing via stac?

**Reply by cmosig** [reply]
https://planetarycomputer.microsoft.com/dataset/sentinel-2-l2a#Example-Notebook mentions more permissive data access with an API key.

**Reply by TomAugspurger** [reply]
Thanks for the link. We'll get those notebooks updated, but there's no need to provide that Subscription Key anymore.

**Reply by cmosig** [reply]
Alright thanks! May I ask what the rate-limit is now? Is there any way of knowing?

**Reply by TomAugspurger** [reply]
I don't recall the exact limit, but the HTTPp response from the `/sas` endpoint will have a `429` status code if you exceed it.

**Reply by tejasnaik0810** [reply]
What's the correct URL to apply for Planetary Computer Account Sign up as the current link is not working ?

**Reply by mmcfarland** [reply]
@tejasnaik0810 We're no longer accepting account requests for the Planetary Computer as the Hub is now deprecated. Previously, access to the Sentinel-1 RTC dataset required an account, but we've removed that restriction. All data and API functionality are available without an account or API Subscription Key (SAS tokens are still required for data access to Azure Blob Storage).

**Reply by tejasnaik0810** [reply]
@mmcfarland Basically I need to run the workflow of forest change detection from farmvibes.ai , for that **pc_key** is required so as per your info whether my workflow will still run without the **pc_key** ?
[ex -"**pc_key**": "@ SECRET(eywa-secrets, pc-sub-key)"]

**Reply by TomAugspurger** [reply]
You can leave that out, or set it to some random value.

**Comment by Alma-Cardenas** [comment]
Thanks for the information guys.

I agree with many of the comments here, the Planetary Computer Hub was a great tool to help democratize the work on Earth Observations and it did lower the barrier of entry, as designed. I was one of the ones who benefited the most on the ease of use as I learned my way through geospatial analysis.  I shall miss the PC Hub.

**Reply by TomAugspurger** [reply]
Thanks Alma!

**Comment by alexgleith** [comment]
I would like to just jump in and say thank you to the MSPC team. All of you!

The architecture, the implementation and the very generous provision of resources has been fantastic and while this is an end of a phase, the fact that the data and APIs are continuing is fantastic. The MSPC is a reference implementation of a whole bunch of tech that is really important, I think, and it will continue to be so.

So, that's it. Thank you!

**Reply by zstatmanweil** [reply]
Seconded!

**Comment by scottyhq** [comment]
Also just wanted to say thanks to everyone who's put so much time and effort into the Hub over the last years! It's been a tremendous resource for education and research. It's sad to see it go, but hopefully Microsoft continues to appreciate the value in maintaining and expanding the data catalog and all of your important contributions to open source software that are essential for bringing the data to life!

**Comment by haojie-w** [comment]
Just want to say thank you as Planetary Computer is a HUGE help to my research. Thank you so much!!! 

I am unfamiliar with Azure service, but given that the hub was running with Azure, is it possible to use paid Azure server and get the same environment setting as the Planetary Computer Hub?

**Reply by songhan89** [reply]
Can refer to this guide

https://planetarycomputer.microsoft.com/docs/concepts/hub-deployment/

**Comment by haojie-w** [comment]
What's the exact time the hub will go offline? Will it be at midnight on 5th June (WET time zone)? 

Still running a few things on Hub and want to make sure pulling everything out in time...

**Reply by lossyrob** [reply]
Hi @whj-stanford, you'll have at least until midnight PT on June 6th to log in. The process for removing the login capability will be applied June 7th.  **(edited: June 7th, not July 7th).**

**Reply by haojie-w** [reply]
Good to know, thank you!

**Comment by lhilarides** [comment]
While I'm producing the latest mangrove deforestation alerts (May 2024) for [Global Mangrove Watch](https://www.globalmangrovewatch.org/) right now, for the last time on the Planetary Computer, I'd like to give a big thank you to the team! The Planetary Computer has been a massive help, enabling us to provide a range of stakeholders with up-to-date data on mangrove deforestation.

**Comment by lossyrob** [comment]
Hi everyone,

We have now retired the Hub.

If you missed the deadline for downloading data, please contact planetarycomputer@microsoft.com, we'll be able to help.

Thank you for all the comments and discussion here. The kind words and thanks are truly appreciated by everyone on the team.

A special thanks to @TomAugspurger , whose hard work and dedication was the primary reason we were able to offer the Hub as long as we could.

Here's to all of you geo data engineers and scientists - thank you for building on top of the Hub, and for continuing to build out this wonderful geospatial community. We can't wait to see what you all create in the future with the Planetary Computer and beyond. 

Cheers,
Rob

**Comment by prasadko** [comment]
[heart]         Prasad Komma reacted to your message:
________________________________

**Comment by ZirconCode** [comment]
Thank you @TomAugspurger and the pchub team for your contributions to the open source ecosystem.

**Comment by Pratichhya** [comment]
Thank you for your platform. It has been very helpful and useful, especially when working with the EO dataset. Especially for many researchers like me. I hope that sharing my experience here, on similar hubs, can help others find alternative solutions, such as the Copernicus Dataspace Ecosystem Hub.

As a user of both platforms, I believe CDSE JupyterLab can serve as a valuable alternative to maintain the efficiency of ongoing projects with smooth transitions. 
 It offers a user-friendly interface for notebooks, code, and data. Connecting users to access and analyse data resources is easy and efficient using APIs available in this Ecosystem, such as openEO, Sentinel Hub, STAC, etc. 

Here is a detailed post on why it could be a good choice. 
https://dataspace.copernicus.eu/news/2024-6-14-transitioning-seamlessly-cdse-jupyterlab-former-ms-planetary-computer-hub-users

**Reply by TomAugspurger** [reply]
Thanks for sharing. Where does that run? Some (most?) users will want to be careful to locate their compute close to the data, especially for workloads where low latency is important or where you're working with a large amount of data.

**Reply by Pratichhya** [reply]
JupyterLab's main acitivities is in Warsaw and JupyterLab's back up activities is in Amsterdam. 
CDSE data is also in Warsaw, so JupyterLab activities is close to the data

**Reply by TomAugspurger** [reply]
Ah, so CDSE hosts data too? Cool. Just a reminder that it's only the Hub that's being retired. People using the Planetary Computer STAC API and datasets we host at scale will want to do that in the same region as the Storage Account (which can be found on the `msft:region` field on the STAC collection).

**Reply by Pratichhya** [reply]
Yes, they are the main Copernicus Data providers along with many other additional collection as shown here: https://dataspace.copernicus.eu/explore-data/data-collections

**Comment by jgomezdans** [comment]
That's very useful, thanks! What about running big processing on top of dask clusters? This was the killer up for me.

(I realise this might be off topic. If so, apologies, but I'm keen on replicating what I had on PC elsewhere!)

Thanks!


________________________________

**Reply by TomAugspurger** [reply]
That's still an option. I typically use [dask-kubernetes](https://kubernetes.dask.org/en/latest/), since I already have AKS clusters handy. https://docs.dask.org/en/latest/deploying.html list some other options, including services like Coiled which would make it as easy (or easier) than the way the Planetary Computer Hub did things.

**Comment by lossyrob** [comment]
> [!NOTE]
> All data from the Planetary Computer Hub will be permanently deleted on the 4th of September 2024.

This is a notice that we will be removing all Hub data permanently on September 4th, 2024.

If you still have not asked for access to your account data, please do so as soon as possible. We will not be able to retrieve any data after the September 4th date.

Thank you.

**Comment by kushankb** [comment]
Hi all,
Does anyone have recommendations on a workaround using other platforms to access data and compute? I know a few different options have been mentioned in this thread. Any favorites? Any progress on the migration guide for Coiled?
Thanks!

**Reply by TomAugspurger** [reply]
> to access data and compute

Just to be clear, the Planetary Computer data and APIs haven't changed. Just the compute (the Planetary Computer Hub).

https://planetarycomputer.microsoft.com/docs/concepts/computing/ has examples of using Planetary Computer data and APIs from the Planetary Computer. Coiled is quite nice too. I suspect that if you create a local environment with the right dependencies (the last one we used is defined at https://github.com/microsoft/planetary-computer-containers/tree/main/python) then everything will work fine.

**Comment by petebunting** [comment]
Hi,

We’ve transitioned notebooks which were using the Planetary Computer hub to coiled - once we worked it out, it was only changing a few lines within the notebook. Found it easier to get working using a docker container rather than environment and created a custom docker container which inherits from pangeo and added the coiled module.

This was the code we changed to get it working:

#########################################################
# Planetary Computer Hub 
#########################################################
cluster = GatewayCluster() # Creates the Dask Scheduler. Might take a minute.
cluster.adapt(minimum=4, maximum=24)
print(cluster.dashboard_link)

client = dask.distributed.Client(cluster, timeout=5)
odc.stac.configure_rio(cloud_defaults=True, client=client)
#########################################################

#########################################################
# Coiled
#########################################################
cluster = coiled.Cluster(
n_workers=4,
region="westeurope",
worker_memory="8 GiB",
container='petebunting/gmw-alerts-contain'
)
client = cluster.get_client()
print(cluster.dashboard_link)
odc.stac.configure_rio(cloud_defaults=True, client=client)
#########################################################

This the whole notebook which is working for us:

https://github.com/globalmangrovewatch/gmw_planetary_computer_s2_alerts/blob/main/02_create_loss_alert_rasters/sen2_monthly_chng_alerts_ndvi_sgl_mnth.ipynb

Cheers, Pete

**Comment by ghidalgo3** [comment]
> [!NOTE]
> All data from the Planetary Computer Hub will be permanently deleted on the 4th of September 2024.

This is a **final** notice that we will be removing all Hub data permanently on September 4th, 2024.

If you still have not asked for access to your account data, please do so as soon as possible. We will not be able to retrieve any data after the September 4th date.

Thank you.

**Comment by ghidalgo3** [comment]
We have deleted all remaining user data from the Hub.
